compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -1101,10 +1153,12 @@ pub fn expand_preparsed_format_args(
         is_literal: parser.is_literal,
     };
 
-    // This needs to happen *after* the Parser has consumed all pieces to create all the spans
+    // This needs to happen *after* the Parser has consumed all pieces to create all the spans.
+    // unverified_pieces is used later to check named argument names are used, so clone each piece.
     let pieces = unverified_pieces
-        .into_iter()
-        .map(|mut piece| {
+        .iter()
+        .map(|piece| {
+            let mut piece = piece.clone();
```

Comment:
This can also be written as

```suggestion
        .iter()
        .cloned()
        .map(|mut piece| {
```

Reply:
@estebank Thanks for the suggestion! Did you want me to fix this? (I wasn't sure since it was in the queue.)

Reply:
There's no need right now if you can't get to it, but keep it in mind for the future. If you do fix it, I can reapprove :)

Reply:
I force pushed the fix -- I hope that was what was expected!